### PR TITLE
Add support for Aetina N510 TX2 Carrier Board

### PR DESCRIPTION
**Pull request by compiaffe**
Adding support for the Aetina N510 carrier board. 

~~This is WIP.~~ It builds, flashes and USB3 ports work as expected.

The BSP is available from here: https://www.aetina.com/dl.php?i=480

It contains a `tegra186-quill-p3310-1000-c03-00-base.dtb`, well as a kernel image. However, Aetina confirmed that they have not modified the kernel, therefore the kernel image can be safely ignored.

I've mirrored the changes in previous PRs to add support for new carrier boards. 
It builds with the resin docker image (see https://forums.balena.io/t/resinos-custom-build-with-docker/1926/5?).
Against my initial expectations,  the `tegra186-quill-p3310-1000-c03-00-base.dtb` does not just simply get copied to the desired location at `/boot/`. As far as I understand, the resin docker image / bitbake creates a new dtb from sources, incorporating the dtb file. Reading on devtalk.nvidia.com [here](https://devtalk.nvidia.com/default/topic/1035622/jetson-tx2/in-system-update-of-dtb-in-r28-2/post/5262418/#5262418) it appears as if the dtb that is build is actually flashed onto a partition from which u-boot will load it. To verify this, during the bitbake compilation encrypted headers are constructed and placed in /opt/tegra-binaries.

---
Associated discussion on: https://forums.balena.io/t/on-jetson-tx2-with-a-custom-build-resin-os-2-19-0-rev1-services-fail-to-install/4207/8

Aetina N510 product page: https://www.aetina.com/products-detail.php?i=234

### CODE REVIEW

**Comment by compiaffe** [comment]
I'm happy to squash and change the commit message as per your conventions once I got it working.

**Comment by compiaffe** [comment]
@floion The USB issue is fixed. It was missing the `ODMDATA` statement in n510-tx2.conf.
Builds, deploys and gives me USB3 as expected.
Thanks for your help!

(BTW congrats and thank you on the rebrand and OS release!)

**Comment by floion** [comment]
Hi @compiaffe , great to see it working.
can you do the commit changes like discussed also in this other PR? https://github.com/balena-os/resin-jetson-tx2/pull/93

**Comment by compiaffe** [comment]
From what I can gather there it is only the following which I already comply with:
- Description less than 72 characters
- Emtpy line after
- Longer description with less than 72 characters per line.

**Comment by floion** [comment]
Hi @compiaffe please see here https://github.com/balena-os/resin-jetson-tx2#pull-requests for details on commit guidelines.
Can you also rebase on latest master please?

**Comment by compiaffe** [comment]
@floion Rebase done and changelog added, too.
The documentation is unclear about the `Signed-off-by` part.
Do I put you and your mail there?

File: layers/meta-resin-jetson/recipes-kernel/linux/linux-tegra_%.bbappend [hunk 1]
```diff
@@ -75,6 +76,7 @@ do_deploy_append() {
     install -m 0600 "${D}/boot/extlinux/extlinux.conf_flasher" "${DEPLOYDIR}"
 
     cp ${WORKDIR}/tegra186-tx2-cti-ASG001-USB3.dtb "${DEPLOYDIR}"
+    cp ${WORKDIR}/tegra186-quill-p3310-1000-c03-00-base.dtb "${DEPLOYDIR}"
```
Comment (compiaffe) [inline layers/meta-resin-jetson/recipes-kernel/linux/linux-tegra_%.bbappend]: I guess if we are doing a subdir to distinquish the file from the base jetson.dtb then I also need to move it here?
```suggestion
    cp ${WORKDIR}/tegra186-quill-p3310-1000-c03-00-base.dtb "${DEPLOYDIR}/n510"
```

File: layers/meta-resin-jetson/recipes-bsp/tegra-binaries/tegra186-flash-dry_28.2.0.bb [hunk 2]
```diff
@@ -23,6 +23,7 @@ SRC_URI = " \
 IMAGE_UBOOT ??= "u-boot-dtb"
 
 KERNEL_DEVICETREE_orbitty-tx2 = "${DEPLOY_DIR_IMAGE}/tegra186-tx2-cti-ASG001-USB3.dtb"
+KERNEL_DEVICETREE_n510-tx2 = "${DEPLOY_DIR_IMAGE}/tegra186-quill-p3310-1000-c03-00-base.dtb"
```
Comment (compiaffe) [inline layers/meta-resin-jetson/recipes-bsp/tegra-binaries/tegra186-flash-dry_28.2.0.bb]:
```suggestion
KERNEL_DEVICETREE_n510-tx2 = "${DEPLOY_DIR_IMAGE}/n510/tegra186-quill-p3310-1000-c03-00-base.dtb"
```

File: layers/meta-resin-jetson/recipes-kernel/linux/linux-tegra_%.bbappend [hunk 3]
```diff
@@ -4,6 +4,7 @@ FILESEXTRAPATHS_append := ":${THISDIR}/${PN}"
 SRC_URI_append = " \
     file://0001-Expose-spidev-to-the-userspace.patch \
     file://tegra186-tx2-cti-ASG001-USB3.dtb \
+    file://tegra186-quill-p3310-1000-c03-00-base.dtb \
```
Comment (compiaffe) [inline layers/meta-resin-jetson/recipes-kernel/linux/linux-tegra_%.bbappend]: If I'm doing the subdir change as mentioned, do I also need to change here?
```suggestion
    file://n510/tegra186-quill-p3310-1000-c03-00-base.dtb \
```

**Comment by floion** [comment]
The Signed-off-by should be that of the commiter, in this case you.
Just make sure when you do your git commits, do `git commit` with the `-s` flag, the signed of by flag. And if git is configured on your machine, it will then add the info to all your commits which you create with the -s flag.

**Comment by floion** [comment]
@resin-jenkins test this please

**Comment by floion** [comment]
@resin-ci retest

File: layers/meta-resin-jetson/recipes-bsp/tegra-binaries/tegra186-flash-dry_28.2.0.bb (hunk 2 above)
Comment (floion) [reply layers/meta-resin-jetson/recipes-bsp/tegra-binaries/tegra186-flash-dry_28.2.0.bb]: Leave the .dtb as you originally put it. I think moving it to a directory won't have the desired effect and anyway it does not hurt to leave it like that

**Comment by floion** [comment]
@resin-ci retest

**Comment by acostach** [comment]
@compiaffe We are currently adding support for LT4 32.2 at: https://github.com/balena-os/balena-jetson/pull/13. If you wish to use it with N510, you can open a PR in that repository with any updates you find necessary for this board (dtb, configs, etc).

**Comment by compiaffe** [comment]
@acostach Thanks for the heads-up - will do.

Are there any major changes on your side that I should look out for or would you expect something similar to this MR to work?

**Pull request merged**